CRM/Contribute/BAO/Contribution.php

```diff
@@ -5202,9 +5203,10 @@ public static function updateMembershipBasedOnCompletionOfContribution($contribu
         // The api assumes num_terms is a special sauce for 'is_renewal' so we need to not pass it when updating a pending to completed.
         // ... except testCompleteTransactionMembershipPriceSetTwoTerms hits this line so the above is obviously not true....
         // @todo once apiv4 ships with core switch to that & find sanity.
-        $membershipParams['num_terms'] = $contribution->getNumTermsByContributionAndMembershipType(
+        $bao = new self();
+        $membershipParams['num_terms'] = $bao->getNumTermsByContributionAndMembershipType(
```

Comment:
Looks to me like `getNumTermsByContributionAndMembershipType` should be changed to a static and then we don't have to do the `new self()`

Reply:
Yeah OK - I was reluctant to get into static vs no static functions in this PR - but this is the only place it's called from so I updated it

Reply:
Thanks. It's cleaner this way and obvious that we're not using anything from the contribution object in that function.